# Explore NCI data

In [12]:
import pandas as pd

## Drug growth

In [52]:
drug_growth = pd.read_csv("../Data/NCI/ComboDrugGrowth_Nov2017.csv", low_memory=False)

In [53]:
drug_growth

COMBODRUGSEQ SCREENER                STUDY    TESTDATE        PLATE  \
0              260496       FG  PZUT00156_86_01_T72  06/23/2011  786-0_1_T72   
1              260497       FG  PZUT00156_86_01_T72  06/23/2011  786-0_1_T72   
2              260498       FG  PZUT00156_86_01_T72  06/23/2011  786-0_1_T72   
3              260499       FG  PZUT00156_86_01_T72  06/23/2011  786-0_1_T72   
4              260500       FG  PZUT00156_86_01_T72  06/23/2011  786-0_1_T72   
...               ...      ...                  ...         ...          ...   
3686470       5732421       1A             0903CB75  03/02/2009          322   
3686471       5732422       1A             0903CB75  03/02/2009          322   
3686472       5732423       1A             0903CB75  03/02/2009          322   
3686473       5732424       1A             0903CB75  03/02/2009          322   
3686474       5732425       1A             0903CB75  03/02/2009          322   

         PANELNBR  CELLNBR PREFIX1    NSC1  SAMPLE1  ...  PERCENTGROWTH  \
0               9       18       S     752       37  ...         85.979   
1               9       18       S     752       37  ...        100.903   
2               9       18       S     752       37  ...         14.147   
3               9       18       S     752       37  ...         71.268   
4               9       18       S     752       37  ...         89.278   
...           ...      ...     ...     ...      ...  ...            ...   
3686470        12       16       S  681239        1  ...        -51.889   
3686471        12       16       S  681239        1  ...        -61.133   
3686472        12       16       S  681239        1  ...        -62.724   
3686473        12       16       S  681239        1  ...        -66.799   
3686474        12       16       S  681239        1  ...        -73.658   

         PERCENTGROWTHNOTZ   TESTVALUE CONTROLVALUE    TZVALUE  \
0                   88.159  332168.000   376781.548  58592.854   
1                  100.763  379656.000   376781.548  58592.854   
2                   27.498  103608.000   376781.548  58592.854   
3                   75.736  285360.000   376781.548  58592.854   
4                   90.945  342664.000   376781.548  58592.854   
...                    ...         ...          ...        ...   
3686470                NaN       0.242        1.636      0.503   
3686471                NaN       0.196        1.636      0.503   
3686472                NaN       0.188        1.636      0.503   
3686473                NaN       0.167        1.636      0.503   
3686474                NaN       0.133        1.636      0.503   

         EXPECTEDGROWTH  SCORE  VALID         PANEL  CELLNAME  
0                90.342    4.0      Y  Renal Cancer     786-0  
1                87.130  -14.0      Y  Renal Cancer     786-0  
2                12.739   -1.0      Y  Renal Cancer     786-0  
3                76.397    5.0      Y  Renal Cancer     786-0  
4                73.681  -16.0      Y  Renal Cancer     786-0  
...                 ...    ...    ...           ...       ...  
3686470             NaN    NaN      Y    CNS Cancer    SF-539  
3686471             NaN    NaN      Y    CNS Cancer    SF-539  
3686472             NaN    NaN      Y    CNS Cancer    SF-539  
3686473             NaN    NaN      Y    CNS Cancer    SF-539  
3686474             NaN    NaN      Y    CNS Cancer   SF-539  

[3686475 rows x 29 columns]

The NSC1 and NSC2 columns contain the IDs of the drugs used for experiments. Infos about the drugs (name and smiles) can be retrieved in the ```ComboCompoundNames_all_with_smiles.csv``` file. The target should be the ```PercentGrowth```column.

Important information such as drug concentrations (conc1 and conc2) should be used as input features (in addition to drug fingerprints). 

For more information about the columns, visit: https://wiki.nci.nih.gov/display/NCIDTPdata/NCI-ALMANAC?preview=/338237347/360284891/ALMANAC%20Data%20Fields.docx

In [54]:
drug_growth[['NSC1', 'NSC2']]

NSC1    NSC2
0           752  3088.0
1           752  3088.0
2           752  3088.0
3           752  3088.0
4           752  3088.0
...         ...     ...
3686470  681239     NaN
3686471  681239     NaN
3686472  681239     NaN
3686473  681239     NaN
3686474  681239     NaN

[3686475 rows x 2 columns]

In [57]:
(drug_growth[['NSC1', 'NSC2']].duplicated() == False).sum()

5460

We have **5460 unique examples of drug pairs**. For some pairs, the experiment has been performed several times

In [60]:
drug_growth['NSC1'].isna().sum()

0

In [61]:
drug_growth['NSC2'].isna().sum()

812961

## Compound infos

In [24]:
compound_names = pd.read_csv("../Data/NCI/ComboCompoundNames_all_with_smiles.csv", index_col=["0"])

In [25]:
compound_names

1  \
0                                 
740                Methotrexate   
740                     Trexall   
740                  Abitrexate   
740                      Mexate   
740                       Folex   
...                         ...   
761432                  XRP6258   
707389        Eribulin mesylate   
737754  Pazopanib hydrochloride   
753082              vemurafenib   
763371              Ruxolitinib   

                                                   smiles  
0                                                          
740     CN(Cc1cnc2nc(N)nc(N)c2n1)c3ccc(cc3)C(=O)N[C@@H...  
740     [Na+].CN(Cc1cnc2nc(N)nc(N)c2n1)c3ccc(cc3)C(=O)...  
740     [Na+].[Na+].CN(Cc1cnc2nc(N)nc(N)c2n1)c3ccc(cc3...  
740     CN(Cc1cnc2nc(N)nc(N)c2n1)c3ccc(cc3)C(=O)N[C@@H...  
740                                    CCCCSP(SCCCC)SCCCC  
...                                                   ...  
761432                                                 -1  
707389                                                 -1  
737754                                                 -1  
753082                                                 -1  
763371                                                 -1  

[1617 rows x 2 columns]

In [23]:
len(compound_names.index.unique())

105

In [22]:
len(compound_names['smiles'].unique())

200

In [44]:
cpt = 0
for idx in compound_names.index.unique():
    has_no_smiles = compound_names.loc[idx]['smiles'] == '-1'
    if type(has_no_smiles) == type(True):
        cpt += 1 - int(has_no_smiles)
    else:
        cpt += 1 - int(has_no_smiles.all())
print(cpt)

93


Out of 105 drugs, 93 of them have a smiles representation. Many of them have more than one smiles representation. We should figure out what to do in such a case.